### DISCUSSION TITLE
MongoDB as SessionStore

### DISCUSSION THREAD

**Discussion by f4bl3d** [maxcountryman/axum-login#150] [Q&A]
Hello!

Sorry for the probably noob question, I'm pretty new with programming and rust ...

If I understood your oauth2 example correctly, one should be able to use mongodb as backend database by simply editing the users.rs and replacing the sql queries with the corresponding mongodb methods. 
My question would be, if I it would be feasible and recommended to use the tower-sessions-mongodb-store crate for the mongodb operations, or should I simply stick with the mongodb crate. 

Thank you!

**Comment by maxcountryman** [comment]
It sounds like you'd like to use Mongo for both your session storage as well as your user storage, is that right?

If so, then yes I'd recommend using the built in Mongo session store for the session manager.

As for the user storage, you should be able to modify the SQLite example like you mentioned: instead of using SQLx, you'll use the Mongo crate.

**Reply by f4bl3d** [reply]
Thank you for your reply!
Yes, you are right, I would like to use mongo for both cases, I'll write the code accordingly.

Thanks!
